File: Models/Category+CoreDataProperties.m
//
//  Category+CoreDataProperties.m
//  ExpenseTracker
//
//  Created by raven on 1/6/26.
//
//

#import "Category+CoreDataProperties.h"

@implementation Category (CoreDataProperties)

+ (NSFetchRequest<Category *> *)fetchRequest {
	return [NSFetchRequest fetchRequestWithEntityName:@"Category"];
}

@dynamic allocatedAmount;
@dynamic createdAt;
@dynamic installmentEndDate;
@dynamic installmentMonths;
@dynamic installmentStartDate;
@dynamic isActive;
@dynamic isIncome;
@dynamic isInstallment;
@dynamic monthlyPayment;
@dynamic name;
@dynamic totalInstallmentAmount;
@dynamic updatedAt;
@dynamic usedAmount;
@dynamic budget;
@dynamic transactions;

@end

File: ExpenseTracker/Sources/Models/Category+CoreDataProperties.swift
//
//  Category+CoreDataProperties.swift
//  ExpenseTracker
//
//  Created by raven on 6/26/25.
//

import Foundation
import CoreData

extension Category {
    
    @nonobjc public class func fetchRequest() -> NSFetchRequest<Category> {
        return NSFetchRequest<Category>(entityName: "Category")
    }
    
    @NSManaged public var allocatedAmount: NSDecimalNumber?
    @NSManaged public var createdAt: Date?
    @NSManaged public var installmentEndDate: Date?
    @NSManaged public var installmentMonths: Int16
    @NSManaged public var installmentStartDate: Date?
    @NSManaged public var isActive: Bool
    @NSManaged public var isIncome: Bool
    @NSManaged public var isInstallment: Bool
    @NSManaged public var monthlyPayment: NSDecimalNumber?
    @NSManaged public var name: String?
    @NSManaged public var totalInstallmentAmount: NSDecimalNumber?
    @NSManaged public var updatedAt: Date?
    @NSManaged public var usedAmount: NSDecimalNumber?
    @NSManaged public var budget: Budget?
    @NSManaged public var transactions: NSSet?
    
}

// MARK: Generated accessors for transactions
extension Category {
    
    @objc(addTransactionsObject:)
    @NSManaged public func addToTransactions(_ value: Transaction)
    
    @objc(removeTransactionsObject:)
    @NSManaged public func removeFromTransactions(_ value: Transaction)
    
    @objc(addTransactions:)
    @NSManaged public func addToTransactions(_ values: NSSet)
    
    @objc(removeTransactions:)
    @NSManaged public func removeFromTransactions(_ values: NSSet)
    
}

extension Category : Identifiable {
    
}
